### ISSUE TITLE
Dynamically created fieldsets not being formatted correctly

### ISSUE THREAD

**Issue by adrianbj** [ryancramerdesign/AdminThemeUikit#58]
Hi Ryan,

This is an unusual one, but it works fine in both the current Default theme and Reno, so I think maybe it should also work in UiKit.

**This is how it should work when you click "Add Another Rule":**

![dynamic fieldset](https://user-images.githubusercontent.com/871211/32472865-8c46dca8-c319-11e7-9b0b-2ed96941ef7c.gif)

**This is what happens with the Uikit theme:**

![dynamic fieldset uikit](https://user-images.githubusercontent.com/871211/32472864-8c31d4de-c319-11e7-8303-9978c1290bc3.gif)

You can test this easily - just install my Custom Upload Names module and attempt to create a new rule. Please excuse the horrible code for creating these extra rules - it's old and borrowed from someone else's module. I should rewrite it, but don't have time.

**Comment by matjazpotocnik** [comment]
Move ajax call to ready().

While there, adjust widths of fields, see https://github.com/ryancramerdesign/AdminThemeUikit/issues/8

**Comment by adrianbj** [comment]
> Move ajax call to ready().

Good catch - thanks, but do you know why it worked in `init() in the Default and Reno themes, but not UiKit?

> While there, adjust widths of fields, see #8

Yeah the widths are not looking as expected. I have a read through Issue #8 and I am not sure I think we should have to modify widths of existing module settings configuration to support UiKit - are we supposed to check what theme is running and adjust widths based on that? Am I missing something? Ryan - can't you add some more logic to handle adjusting widths if the user is running UiKit?

**Comment by matjazpotocnik** [comment]
Because AdminThemeUikit is hooking before Inputfield::render and Inputfield::renderValue to add required uk-* classes and those hooks were not fired in init().

I'm not sure if Ryan is going to add some logic there, so right now you have to adjust widths...

**Comment by adrianbj** [comment]
> Because AdminThemeUikit is hooking before Inputfield::render and Inputfield::renderValue to add required uk-* classes and those hooks were not fired in init().

Thanks - sorry I should have investigated myself :) It does seem like this might affect other modules and general site hooks - could be a gotcha for others.

> I'm not sure if Ryan is going to add some logic there, so right now you have to adjust widths...

I just had a good look at the table you posted in that other issue, but I don't honestly see this as a good solution - having to match those allowed percentages will break lots of existing setups. In my opinion this is a step backwards unless it can be dealt with automatically. Or maybe the entire percentage approach needs to be deprecated and move to 1, 1/2, 1/3, 1/4, 1/5, 1/6 approach for column widths. Any thoughts Ryan?

**Comment by adrianbj** [comment]
> Going along the road with 1/x approach for all themes would probably cause problems to "old" admin themes.

I was thinking that maybe the old themes could be updated to support a 1/x approach because I don't see how UiKit is going to support percentages without adding automatic conversion to 1/x. Anyway, as you said, let's see what Ryan has to say.

**Comment by ryancramerdesign** [comment]
Percentage are supported in AdminThemeUikit (they have to be) but it maps them to uk-width classes. When there isn't an exact match for the percentage, it falls back to the closest matching uk-width class. It sounds like maybe I don't have the mapping quite right for a particular case. Can you tell me what the specified width (columnWidth) of the columns are where it's breaking the table?

**Comment by adrianbj** [comment]
In this case I am using:
19, 27, 27, 27, 25, 25, 25, 25
